Provenance-Emu/Provenance
# XCode15 / iOS17: Add 3DS Support through EmuThreeds / Update Play! PS2 to latest code

### What does this PR do
   * Added emuthreed to play 3DS games 
   * Added 3DS touch screen controller
   * Update Play! to latest GitHub (provides improved sound compared to before)
   * Prevent game from being moved into conflicts folder during artwork download
   * Added RxRealm as library with XCode15 Beta6 patches
   * Update Flycast to latest GitHub tree, with Vulkan support
   * Added Sound Volume Control Option to RetroArch, Emuthreeds, Play, etc
   * LZH / RAR / ZIP archive load support for NP2Kai to play PC98 games with boot disk from archive (extracted files saved to Battery States)
   * Zip archive load / Play support for DosBox / NeoGeo / MAME
   * Optimized ROM import w/ Caching in RomDatabase, added 'Refresh' button to scan / update the game library w/ new rom
   * Various updates to GameLibrary interfaces to work / appear ok on iOS17
   * Added Apple IIe core support via MAME
   * Added MAME / NeoGeo touchscreen controller
   * Vulkan is now the default graphics renderer on PPSSPP
   

On iOS17 AltKit does not work, so to enable JIT:
  1. Launch Provenance XL Release on iPhone/iPad, Launch XCode15, select the iPhone/iPad
  2. Debug -> Attach to Process by PID or Name
  3. Enter "Provenance XL" (assuming Provenance XL Release is installed), Click Attach. After confirming the game (PS2, Wii, etc) is working, Debug -> Detach from Provenance XL

### Where should the reviewer start
emuthreed (should play) -> Play!

### How should this be manually tested
Play .3DS games with emuthreed, Play! offers more games compatibility

### Any background context you want to provide

### What are the relevant tickets
https://github.com/Provenance-Emu/Provenance/issues/2201

### Screenshots (important for UI changes)
![IMG_0930](https://github.com/Provenance-Emu/Provenance/assets/6903427/5808b40f-c1da-4ccb-9434-8ce86abbf2ad)
![IMG_0934](https://github.com/Provenance-Emu/Provenance/assets/6903427/18b28f08-0827-4853-b0bf-bda4681780af)
![IMG_0936](https://github.com/Provenance-Emu/Provenance/assets/6903427/9c8e0dd2-a9ca-4af8-8c36-4e548e6f0b9c)

### Questions
